### ISSUE TITLE
object information export (with plugins) runs into hdf5 header size limit

### ISSUE THREAD

**Issue by k-dominik** [ilastik/ilastik#2714]
## Describe the bug

There is a [limit to the hdf5 header length for datasets (64kb)](https://davis.lbl.gov/Manuals/HDF5-1.8.7/UG/13_Attributes.html) (header itself is an attribute), which can be reached with the structured array types we're saving the object features in. I've read different accounts on how many columns one can save, but it depends on column names and dtypes.

## Error message/traceback

If applicable, please add tracebacks with formatting, such as

```pytb
ERROR 2023-05-23 12:05:18,292 log_exception 5138 12942405632 Export failed
ERROR 2023-05-23 12:05:18,293 log_exception 5138 12990263296 Traceback (most recent call last):
  File "/.../ilastik/ilastik/applets/dataExport/dataExportGui.py", line 429, in exportSync
    self.parentApplet.post_process_lane_export(lane_index)
  File "/.../ilastik/ilastik/applets/objectClassification/objectClassificationDataExportApplet.py", line 63, in post_process_lane_export
    req = self._tableExporter.export_object_data(lane_index, show_gui=False).wait()
  File "/.../ilastik/lazyflow/request/request.py", line 541, in wait
    return self._wait(timeout)
  File "/.../ilastik/lazyflow/request/request.py", line 564, in _wait
    self._wait_within_foreign_thread(timeout)
  File "/.../ilastik/lazyflow/request/request.py", line 624, in _wait_within_foreign_thread
    raise_with_traceback(exc_value, exc_tb)
  File "/.../mambaforge/envs/ilastik/lib/python3.7/site-packages/future/utils/__init__.py", line 446, in raise_with_traceback
    raise exc.with_traceback(traceback)
  File "/.../ilastik/lazyflow/request/request.py", line 360, in _execute
    self._result = self.fn()
  File "/.../ilastik/ilastik/applets/objectClassification/opObjectClassification.py", line 185, in do_export
    export_file.write_all(settings["file type"], settings["compression"])
  File "/.../ilastik/ilastik/utility/exportFile.py", line 457, in write_all
    compression if compression is not None else {},
  File "/.../ilastik/ilastik/utility/exportFile.py", line 493, in _make_h5_dataset
    dset = fout.create_dataset(table_name, sanitized_table.shape, data=sanitized_table, **compression)
  File "/.../mambaforge/envs/ilastik/lib/python3.7/site-packages/h5py/_hl/group.py", line 161, in create_dataset
    dsid = dataset.make_new_dset(group, shape, dtype, data, name, **kwds)
  File "/.../mambaforge/envs/ilastik/lib/python3.7/site-packages/h5py/_hl/dataset.py", line 156, in make_new_dset
    dset_id = h5d.create(parent.id, name, tid, sid, dcpl=dcpl, dapl=dapl)
  File "h5py/_objects.pyx", line 54, in h5py._objects.with_phil.wrapper
  File "h5py/_objects.pyx", line 55, in h5py._objects.with_phil.wrapper
  File "h5py/h5d.pyx", line 87, in h5py.h5d.create
ValueError: Unable to create dataset (object header message is too large)


```
CC: @oanegros

**Comment by oanegros** [comment]
It seems that ilastik is not using the latest HDF5 spec, as large attribute storage was introduced in HDF5 1.8 and adding `libver='latest'` "/.../ilastik/ilastik/utility/exportFile.py", line 450, in _make_h5_dataset :
`with h5py.File(self.file_name, "w", libver='latest') as fout:`
seems to fix this. However, i don't know how much this may reduce the compatibility (and maybe it might be a good idea to go to ` libver=('earliest', 'v108')`)
h5py version bounding docs: https://docs.h5py.org/en/stable/high/file.html

**Comment by k-dominik** [comment]
fixed via #2746
